torch/_refs/__init__.py

```diff
@@ -4647,6 +4648,48 @@ def triu_indices(
     )
 
 
+@register_decomposition(torch.ops.aten.bucketize)
+@out_wrapper()
+def bucketize(
+    a: TensorLikeType,
+    boundaries: TensorLikeType,
+    *,
+    out_int32: bool = False,
+    right: bool = False,
+):
+
+    out_dtype = torch.int32 if out_int32 else torch.int64
+    n_boundaries = boundaries.shape[-1]
+
+    # can't use int32 as indexes, so we have to do all computations with int64 and convert at the end
+    start = torch.zeros(a.shape, device=a.device, dtype=torch.int64)
+    end = start + n_boundaries
+    # Max depth of the binary search
+    niters = int(math.log2(n_boundaries)) + 1
+    if not (right):
+        # equivalent to std::upper_bound
+        for _ in range(niters):
+            cond_update = start < end
```

Comment:
I understand this is probably written this way for readability, but could the algorithm be a little more clever and initialize cond_update as a tensor full of Falses without performing this computation?

Reply:
Not sure if that will actually matter much in practice, but it is easy enough to do and doesn't make the code that much uglier so I added it.